webapp/channels/src/components/resizable_sidebar/resizable/index.tsx

```diff
@@ -0,0 +1,317 @@
+        setIsResizeLineSelected(true);
+        setDir(dir);
+
+        toggleColResizeCursor();
+
+        if (onResizeStart) {
+            onResizeStart();
+        }
+    }, [onResizeStart]);
+
+    const handleMouseMove = useCallback(requestAnimationFrameForMouseMove((e: MouseEvent) => {
+        const wrapperElem = wrapperRef.current;
+
+        if (!previousClientX.current || !wrapperElem) {
+            return;
+        }
+
+        if (!isResizeLineSelected) {
+            return;
+        }
+
+        e.preventDefault();
+
+        const resizeLine = dir === ResizeDirection.LEFT ? leftResizeLineRef.current : rightResizeLineRef.current;
+
+        if (!resizeLine) {
+            return;
+        }
+
+        const prevWidth = wrapperElem?.getBoundingClientRect().width ?? 0;
+        let widthDiff = 0;
+
+        switch (dir) {
+        case ResizeDirection.LEFT:
+            widthDiff = e.clientX - previousClientX.current;
+            break;
+        case ResizeDirection.RIGHT:
+            widthDiff = previousClientX.current - e.clientX;
+            break;
+        }
+
+        const newWidth = prevWidth + widthDiff;
+
+        if (isOverLimit(newWidth, maxWidth, minWidth)) {
+            return;
+        }
+
+        if (resizeLine.classList.contains('snapped')) {
+            const diff = newWidth - defaultWidth;
+
+            if (isSizeLessThanSnapSize(diff)) {
+                return;
+            }
+
+            if (onResize) {
+                onResize(newWidth);
+            }
+
+            setWidth(wrapperElem, newWidth);
+
+            resizeLine.classList.remove('snapped');
+            previousClientX.current = e.clientX;
+
+            return;
+        }
+
+        previousClientX.current = e.clientX;
+
+        const shouldSnap = shouldSnapWhenSizeGrown(newWidth, prevWidth, defaultWidth) || shouldSnapWhenSizeShrunk(newWidth, prevWidth, defaultWidth);
+
+        if (isSnapableSpeed(widthDiff) && shouldSnap) {
+            if (onResize) {
+                onResize(defaultWidth);
+            }
+
+            switch (dir) {
+            case ResizeDirection.LEFT:
+                previousClientX.current += defaultWidth - newWidth;
+                break;
+            case ResizeDirection.RIGHT:
+                previousClientX.current += newWidth - defaultWidth;
+                break;
+            }
+
+            setWidth(wrapperElem, defaultWidth);
+
+            resizeLine.classList.add('snapped');
+            return;
+        }
+
+        setWidth(wrapperElem, newWidth);
+        if (onResize) {
+            onResize(newWidth);
+        }
+    }), [isResizeLineSelected, maxWidth, minWidth, dir]);
+
+    const handleMouseUp = useCallback(() => {
+        setIsResizeLineSelected(false);
+        setDir(null);
+
+        toggleColResizeCursor();
+
+        if (onResizeEnd) {
+            onResizeEnd();
+        }
+    }, [onResizeEnd]);
+
+    useEffect(() => {
+        if (!isResizeLineSelected || !dir || disabled) {
```

Comment:
@kyeongsoosoo What is the benefit of this if statement?
cc// @M-ZubairAhmed @hmhealey

Reply:
To prevent unnecessary event registration, the event was registered at the moment of pressing the resize line